universal-application-tool-0.0.1/app/controllers/LoginController.java

```diff
@@ -0,0 +1,50 @@
+package controllers;
+
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.pac4j.core.context.session.SessionStore;
+import org.pac4j.core.exception.http.RedirectionAction;
+import org.pac4j.core.http.adapter.HttpActionAdapter;
+import org.pac4j.oidc.client.OidcClient;
+import org.pac4j.play.PlayWebContext;
+import org.pac4j.play.http.PlayHttpActionAdapter;
+import play.mvc.Controller;
+import play.mvc.Http;
+import play.mvc.Result;
+
+/**
+ * These routes can be hit even if you are logged in already, which is what allows the merge logic -
+ * normally you need to be logged out in order to be redirected to a login page.
+ */
+public class LoginController extends Controller {
+  @Inject
+  @Named("idcs")
+  OidcClient idcsClient;
+
+  @Inject
+  @Named("ad")
+  OidcClient adClient;
+
+  @Inject SessionStore sessionStore;
```

Comment:
question: could we move these to a constructor to be more consistent with our other dependency injection?